ci/util.py

```diff
@@ -262,24 +266,43 @@ def _count_elements(value, count=0, max_elements_count=100000):
 
 def lint_yaml_file(path, linter_config: dict={'extends': 'relaxed'}):
     existing_file(path)
-    cfg = yamllint.config.YamlLintConfig(yaml.dump(linter_config))
     info(f'linting YAML file: {path}')
-    worst_level = 0
+
     with open(path) as f:
-        for problem in yamllint.linter.run(f, conf=cfg, filepath=path):
-            # problem is of type yamllint.linter.LintProblem
-            level = problem.level
-            if isinstance(problem.level, str):
-                level = yamllint.linter.PROBLEM_LEVELS[problem.level]
-
-            if level == yamllint.linter.PROBLEM_LEVELS['warning']:
-                warning(str(problem))
-            elif level > yamllint.linter.PROBLEM_LEVELS['warning']:
-                error(str(problem))
-
-            worst_level = max(worst_level, level)
-    if worst_level >= yamllint.linter.PROBLEM_LEVELS['error']:
-        fail('found errors whilst linting (see above)')
+        lint_yaml(f.read(), linter_config)
+
+
+def _parse_yamllint_problem_level(problem):
+    # Yamllint uses a dict with mappings for both int->level and level-> int. Consistently use
+    # the int representation
+    if isinstance(problem.level, str):
+        return yamllint.linter.PROBLEM_LEVELS[problem.level]
+    else:
+        return problem.level
+
+
+def lint_yaml(input, config={'extends': 'relaxed'}):
```

Comment:
i think the processing / conversion of yamllint's results should be separated into a dedicated function, if you are already refactoring it. this function could return a dataclass, or maybe a dict with findings grouped by severity

lint_yaml, and lint_yaml_file would then instrument this, and - depending on the result - raise.